tools/lsp/main.rs

```diff
@@ -72,8 +72,12 @@ impl RequestHolder {
             }
         };
 
-        let result = f(param)?;
-        self.1.send(Message::Response(Response::new_ok(id, result)))?;
+        match f(param) {
+            Ok(r) => self.1.send(Message::Response(Response::new_ok(id, r)))?,
+            Err(e) => {
+                self.1.send(Message::Response(Response::new_err(id, 23, format!("{}", e))))?
```

Comment:
What’s 23?:)

Does the error implement to_string() or have description()?

Reply:
You should enjoy the evening:-)

I did not want to change all the Errors used all over the place to provide some extra exit code. I find that has little value overall. So I went for a nice i32.

`Error::description()` is deprecated since Rust 1.42 and you should use the `Display` trait  instead.